src/nbl/asset/utils/CSPIRVIntrospector.cpp

```diff
@@ -13,472 +13,528 @@ namespace nbl::asset
+
+        const spirv_cross::SPIRType& type = comp.get_type(entry.selfTypeID);
+
+        type_ptr<true> pType;
+        auto found = typeCache.find(entry.selfTypeID);
+        if (found!=typeCache.end())
+            pType = found->second;
+        else
         {
-            if (io.type == ESIT_STAGE_INPUT)
-            {
-                auto& attr = vtxInput.attributes[io.location];
-                attr.binding = io.location; //assume attrib number = binding number
-                attr.format = glslType2E_FORMAT(io.glslType.basetype, io.glslType.elements);
-                attr.relativeOffset = reloffset;
+            const auto memberCount = type.member_types.size();
+            pType = addType(memberCount);
+            if (type.basetype==spirv_cross::SPIRType::Struct)
+            for (uint32_t i=0; i<memberCount; i++)
+                introspectionStack.emplace(type.member_types[i],entry.selfTypeID,pType,i);
+        }
+        // pointer might change after allocation of something new, so always access through this lambda
+        auto getTypeStore = [&]()->SType<true>*{return pType(m_memPool.data());};
+        
+        if (entry.isRoot())
+            root->type = pType;
+        else
+        {
+            auto getParentTypeStore = [&]()->auto {return entry.parentType(m_memPool.data());};
+            getParentTypeStore()->memberTypes()(m_memPool.data())[entry.memberIndex] = pType;
+            getParentTypeStore()->memberNames()(m_memPool.data())[entry.memberIndex] = addString(comp.get_member_name(entry.parentTypeID,entry.memberIndex));
+            const auto& parentType = comp.get_type(entry.parentTypeID);
+            getParentTypeStore()->memberSizes()(m_memPool.data())[entry.memberIndex] = comp.get_declared_struct_member_size(parentType,entry.memberIndex);
+            getParentTypeStore()->memberOffsets()(m_memPool.data())[entry.memberIndex] = comp.type_struct_member_offset(parentType,entry.memberIndex);
+            getParentTypeStore()->memberStrides()(m_memPool.data())[entry.memberIndex] = getTypeStore()->isArray() ? comp.type_struct_member_array_stride(parentType,entry.memberIndex):0u;
+//            comp.get_declared_struct_size();
+//            comp.get_declared_struct_size_runtime_array();
+#if 0 
+    for (uint32_t m = 0u; m < memberCnt; ++m)
+    {
+        MembT& member = dstMembers.array[m];
+        const spirv_cross::SPIRType& mtype = comp.get_type(allMembersTypes[m]);
 
-                //all formats returned by glslType2E_FORMAT() already are multiple-of-4 bytes so no need to pad
-                reloffset += getTexelOrBlockBytesize(static_cast<E_FORMAT>(attr.format));
+        member.rowMajor = _comp.get_member_decoration(_parentType.self, m, spv::DecorationRowMajor); // ?
 
-                vtxInput.enabledAttribFlags |= (1u << io.location);
-            }
+            //if (member.mtxColCnt > 1u)
+            //    member.mtxStride = _comp.type_struct_member_matrix_stride(_parentType, m);
+#endif
```

Comment:
where do you handle the matrix info?

Reply:
nowhere yet

Reply:
they need to be done same way as `memberSizes`, offsets and strides